application_sdk/common/observability.py

```diff
@@ -0,0 +1,202 @@
+    LOG_RETENTION_DAYS,
+    OBJECT_STORE_NAME,
+    STATE_STORE_NAME,
+)
+
+
+class AtlanObservability:
+    """Base class for Atlan observability."""
+
+    def __init__(self):
+        """Initialize the observability base class."""
+        # Initialize instance variables
+        self._log_buffer = []
+        self._buffer_lock = threading.Lock()
+        self._last_flush_time = time()
+        self._batch_size = LOG_BATCH_SIZE
+        self._flush_interval = LOG_FLUSH_INTERVAL
+        self.parquet_path = os.path.join(LOG_DIR, LOG_FILE_NAME)
+
+        # Ensure log directory exists
+        os.makedirs(LOG_DIR, exist_ok=True)
+
+    async def parquet_sink(self, message: Any):
+        try:
+            log_record = {
+                "timestamp": message.record["time"].timestamp(),
+                "level": message.record["level"].name,
+                "logger_name": message.record["extra"].get("logger_name", ""),
+                "message": message.record["message"],
+                "file": str(message.record["file"].path),
+                "line": message.record["line"],
+                "function": message.record["function"],
+            }
+            for key, value in message.record["extra"].items():
+                if key not in log_record:
+                    log_record[key] = value
+
+            with self._buffer_lock:
+                self._log_buffer.append(log_record)
+                now = time()
+                if (
+                    len(self._log_buffer) >= self._batch_size
+                    or (now - self._last_flush_time) >= self._flush_interval
+                ):
+                    self._last_flush_time = now
+                    buffer_copy = self._log_buffer[:]
+                    self._log_buffer.clear()
+                else:
+                    buffer_copy = None
+
+            if buffer_copy is not None:
+                await self._flush_records(buffer_copy)
+        except Exception as e:
+            logging.error(f"Error buffering log: {e}")
+
+    async def _flush_buffer(self, force=False):
+        with self._buffer_lock:
+            if self._log_buffer:
+                buffer_copy = self._log_buffer[:]
+                self._log_buffer.clear()
+            else:
+                buffer_copy = None
+        if buffer_copy:
+            await self._flush_records(buffer_copy)
+
+    async def _flush_records(self, records):
+        try:
+            df = pd.DataFrame(records)
+            if os.path.exists(self.parquet_path):
+                existing_table = pq.read_table(self.parquet_path)
+                existing_df = existing_table.to_pandas()
+                df = pd.concat([existing_df, df], ignore_index=True)
+            table = pa.Table.from_pandas(df)
+            pq.write_table(table, self.parquet_path, compression="snappy")
+            with open(self.parquet_path, "rb") as f:
+                file_content = f.read()
+                filename = "logs/log.parquet"
```

Comment:
I believe the file name and path are configured via env vars? if so, let's make them consistent